query-engine/connectors/mongodb-query-connector/src/root_queries/raw.rs

```diff
@@ -0,0 +1,290 @@
+use mongodb::{
+    bson::{from_bson, Bson, Document},
+    options::*,
+};
+use std::convert::TryFrom;
+
+use crate::{error::MongoError, BsonTransform};
+
+impl TryFrom<Document> for MongoCommand {
+    type Error = MongoError;
+
+    fn try_from(doc: Document) -> Result<Self, Self::Error> {
+        if doc.contains_key("find") {
+            return MongoCommand::try_into_find(doc);
+        }
+
+        if doc.contains_key("aggregate") {
+            return MongoCommand::try_into_aggregate(doc);
+        }
+
+        if doc.contains_key("findAndModify") {
+            return MongoCommand::try_into_find_and_modify(doc);
+        }
+
+        if doc.contains_key("count") {
+            return MongoCommand::try_into_count(doc);
+        }
+
+        if doc.contains_key("distinct") {
+            return MongoCommand::try_into_distinct(doc);
+        }
+
+        if doc.contains_key("insert") {
+            return MongoCommand::try_into_insert_many(doc);
+        }
+
+        // If there is no top-level driver API for a command or the driver doesn't allow expressing everything that can be done via the raw command
+        // just try executing it as-is with `run_command`
+        Ok(MongoCommand::new_raw(doc))
+    }
+}
+
+#[allow(clippy::large_enum_variant)]
+pub enum MongoCommand {
+    Raw {
+        cmd: Document,
+    },
+    Handled {
+        collection: String,
+        operation: MongoOperation,
+    },
+}
```

Comment:
Tough call here, not sure it matters much anyway. Point being that the `Handled` variant is much larger than `Raw`. Clippy suggested to box `MongoOperation`, only if `Raw` is used as often as Handled or even more (because memory-wise, it'll always alloc the size of `Handled` since it's the biggest variant).

I judged that `Handled` would likely be used more often than `Raw` given that it handles `aggregate`, but honestly we might prefer just boxing `MongoOperation` as I/we don't really know.

Reply:
Good thoughts. I think it's not relevant: We're talking about 1 allocation per raw request.